server/src/main/java/org/elasticsearch/cluster/metadata/MetaDataUpdateSettingsService.java

```diff
@@ -187,6 +188,14 @@ public ClusterState execute(ClusterState currentState) {
                     }
                 }
 
+                // increment settings versions
+                for (final String index : actualIndices) {
+                    if (same(currentState.metaData().index(index).getSettings(), metaDataBuilder.get(index).getSettings()) == false) {
+                        final IndexMetaData.Builder builder = IndexMetaData.builder(metaDataBuilder.get(index));
+                        builder.settingsVersion(1 + builder.settingsVersion());
+                        metaDataBuilder.put(builder);
```

Comment:
can we instead increment the version at lines 169 and 185? like: 

```
metaDataBuilder.put(IndexMetaData.builder(indexMetaData).settings(finalSettings).settingsVersion(1 + builder.settingsVersion()));
```

At these places we do know that the index versions have been updated. And then we don't have to check whether settings have been changed in another for loop.

Reply:
That would not be correct since a settings update does not necessarily mutate any settings. The test `UpdateSettingsIT#testSettingsVersionUnchanged` test for this case. That is, a settings update that does not change any settings should not modify the settings version.

Reply:
thanks for adding that test!